homeassistant/components/hassio/__init__.py

```diff
@@ -286,13 +301,18 @@ def get_supervisor_ip():
     return os.environ["SUPERVISOR"].partition(":")[0]
 
 
-async def async_setup(hass, config):
+async def async_setup(hass: HomeAssistant, config: Config) -> bool:
     """Set up the Hass.io component."""
     # Check local setup
     for env in ("HASSIO", "HASSIO_TOKEN"):
         if os.environ.get(env):
             continue
         _LOGGER.error("Missing %s environment variable", env)
+        config_entries = hass.config_entries.async_entries(DOMAIN)
+        if config_entries:
+            hass.async_create_task(
```

Comment:
This should be outside the for loop, it will now run twice if both are missing.

Reply:
It shouldn't because we return False at the end if we can't find the current env var

Reply:
Right :+1: